common/src/disk.rs

```diff
@@ -126,6 +127,21 @@ impl DatasetName {
         }
     }
 
+    /// Returns the mountpoint of the dataset.
+    ///
+    /// If this dataset should be encrypted, this automatically adds the
+    /// "crypt" dataset component.
+    pub fn mountpoint(&self, root: &Utf8Path) -> Utf8PathBuf {
```

Comment:
I appreciate you moving this function here, but to avoid further confusion, could this also take a `bool: zoned` parameter? (the logic for how to handle this can be moved up from `sled-storage/src/manager.rs` too)

here are some example datasets / their mountpoints from dogfood:

```
oxp_bf428719-1b16-4503-99f4-ad95846d916f/crypt                                                                /pool/ext/bf428719-1b16-4503-99f4-ad95846d916f/crypt
oxp_bf428719-1b16-4503-99f4-ad95846d916f/crypt/debug                                                          /pool/ext/bf428719-1b16-4503-99f4-ad95846d916f/crypt/debug
oxp_bf428719-1b16-4503-99f4-ad95846d916f/crypt/zone                                                           /pool/ext/bf428719-1b16-4503-99f4-ad95846d916f/crypt/zone
oxp_bf428719-1b16-4503-99f4-ad95846d916f/crypt/zone/oxz_crucible_7153983f-8fd7-4fb9-92ac-0f07a07798b4         /pool/ext/bf428719-1b16-4503-99f4-ad95846d916f/crypt/zone/oxz_crucible_7153983f-8fd7-4fb9-92ac-0f07a07798b4
oxp_cf940e15-dbc5-481b-866a-4de4b018898e/crucible                                                             /data
```

The "durable data for crucible" -- which gets mounted inside the zone -- has a mountpoint of `/data`, which is our default for datasets where `zoned=true`.

Reply:
Definitely yeah. I think it doesn't need to take a parameter though, because the value of `zoned` depends on the `DatasetKind` variant in use. But I'll move that logic up as well.